rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -77,6 +77,47 @@ lv2_fs_mount_point* lv2_fs_object::get_mp(std::string_view filename)
 	return &g_mp_sys_dev_hdd0;
 }
 
+std::size_t lv2_fs_object::count_path_items(std::string_view filename)
+{
+	std::size_t count = 0;
+
+	while (true)
+	{
+		auto begin = filename.find_first_not_of('/');
+
+		if (begin == umax)
+		{
+			break;
+		}
+
+		const auto end = filename.find_first_of('/', begin);
+
+		if (filename.compare(begin, end, "."sv) == 0)
+		{
+			// Nothing
+		}
+		else if (filename.compare(begin, end, ".."sv) == 0)
+		{
+			// Stop decreasing after 0
+			// TODO: ENOENT in the case of trying to decrease after 0 (except for /app_home)
+			if (count) count--;
+		}
+		else
+		{
+			count++;
+		}
+
+		if (end == umax)
```

Comment:
Should this not also be string::npos?
And shouldn't we be testing for this before passing it into the compare?
Also.. for these compares, do we expect leading '/' symbols? or will the filename be precisely "." or ".." in which case we could use the simple string equality operator (==)

Reply:
theres nothing wrong with passing size_t(-1) to substr end which is the same as compare end position, its a feature and behaves like passing filename.size() - begin.

Reply:
it analyzes each path element alone. "." and ".." are not string terminating or something so more path elements can be followed agter them.